packages/modular-scripts/src/utils/getPackageDependencies.ts

```diff
@@ -0,0 +1,83 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { Project } from 'ts-morph';
+import type { CoreProperties } from '@schemastore/package';
+import getModularRoot from './getModularRoot';
+import getLocation from './getLocation';
+import getWorkspaceInfo from './getWorkspaceInfo';
+
+type DependencyManifest = NonNullable<CoreProperties['dependencies']>;
+
+const npmPackageMatcher =
+  /^(@[a-z0-9-~][a-z0-9-._~]*\/)?[a-z0-9-~][a-z0-9-._~]*/;
+
+/* Get dependencies from import / require declarations, since they could be hoisted to the root workspace. Exclude test files. */
+function getDependenciesFromSource(workspaceLocation: string) {
+  const project = new Project();
```

Comment:
Does this need a working directory?

Reply:
No, you just add the files you want to act on with `addSourceFilesAtPaths`. You can also point it to a `tsconfig`, but I found it gets confused with workspaces.